src/dev/generate_release_note.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { resolve } from 'path';
+import { readFileSync, writeFileSync, readdirSync, unlinkSync } from 'fs';
+import { load as loadYaml } from 'js-yaml';
+import { readdir } from 'fs/promises';
+import { version as pkgVersion } from '../../package.json';
+import {
+  validateFragment,
+  getCurrentDateFormatted,
+  Changelog,
+  SECTION_MAPPING,
+  fragmentDirPath,
+  SectionKey,
+  releaseNotesDirPath,
+  filePath,
+} from './generate_release_note_helper';
+
+// Function to add content after the 'Unreleased' section in the changelog
+function addContentAfterUnreleased(path: string, newContent: string): void {
+  let fileContent = readFileSync(path, 'utf8');
+  const targetString = '## [Unreleased]';
+  const targetIndex = fileContent.indexOf(targetString);
+
+  if (targetIndex !== -1) {
+    const endOfLineIndex = fileContent.indexOf('\n', targetIndex);
+
+    if (endOfLineIndex !== -1) {
+      fileContent =
+        fileContent.slice(0, endOfLineIndex + 1) +
+        '\n' +
+        newContent +
+        fileContent.slice(endOfLineIndex + 1);
+    } else {
+      throw new Error('End of line for "Unreleased" section not found.');
+    }
+  } else {
+    throw new Error("'## [Unreleased]' not found in the file.");
+  }
+
+  writeFileSync(path, fileContent);
+}
+
+const currentDate = getCurrentDateFormatted();
+
+// Initialize sections
+const sections: Changelog = (Object.fromEntries(
+  Object.keys(SECTION_MAPPING).map((key) => [key, []])
+) as unknown) as Changelog;
```

Comment:
Why don't we make sections an internal variable of `readFragments` and let `readFragments` return sections?

Reply:
changed, thanks!